flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/connection/JdbcConnectionProvider.java

```diff
@@ -20,13 +20,40 @@
 import org.apache.flink.annotation.Internal;
 
 import java.sql.Connection;
+import java.sql.SQLException;
 
 /**
  * JDBC connection provider.
  */
 @Internal
 public interface JdbcConnectionProvider {
-	Connection getConnection() throws Exception;
+	/**
+	 * Get existing connection.
+	 *
+	 * @return existing connection
+	 */
+	Connection getExistingConnection();
```

Comment:
Is this method necessary? Can users use `getConnection` instead?

Reply:
This method currently exist for reasons:
* Bypass java checked exception restriction and serve as drop-in replacement for old `connection` field in migrating classes.
* Test access with no side effect to wrapped connection in `JdbcInputFormat.getDbConn`, `AbstractJdbcOutputFormat.getConnection`, `JdbcLookupFunction.getDbConnection` and `JdbcRowDataLookupFunction.getDbConnection` without resort to cast or additional `VisibleForTesting`.

Reply:
That makes sense. Then I would suggest to add `@Nullable` annotation on the return type or check null in the implementation.

Reply:
`@Nullable` added.